docs/setup/telemetry.mdx

```diff
@@ -0,0 +1,113 @@
+
+### Tracing
+
+Tracing is provided through the `Tracer` class which is a wrapper around OpenTelemetry's tracing capabilities. It allows you to:
+
+- Create spans to measure operation execution time
+- Add attributes and events to spans
+- Establish parent-child relationships between spans
+- Link spans across asynchronous operations
+
+Saleor implements two distinct scopes for traces:
+- **CORE**: For internal system operations and core functionality
+- **SERVICE**: For business logic and service-level operations
+
+What is being traced:
+
+- **HTTP requests** including their duration, full URL, request method, client's IP address, and length of response.
+- **GraphQL queries** including their duration, query string, and execution errors.
+- **GraphQL resolvers** including their duration, field name, parent type, and execution errors.
+- **Database queries** including their duration and the SQL statement.
+
+### Metrics
+
+Metrics are handled through the `Meter` class, which is a wrapper around OpenTelemetry's metrics capabilities. It supports:
+
+- Counters
+- Histograms
+- Up/Down Counters
+
+Like traces, metrics are also separated into CORE and SERVICE scopes.
+
+### Context propagation
+
+Saleor implements context propagation to ensure telemetry context is maintained across:
+- Synchronous operations
+- Asynchronous Celery tasks (via [Span Links](https://opentelemetry.io/docs/concepts/signals/traces/#span-links))
+- Web requests
+
+Context propagation between Saleor and external services is handled through OpenTelemetry's [W3C Trace Context](https://www.w3.org/TR/trace-context/) standard.
+
+## Setting up your own observability
+
+If you're running your own Saleor instance, you can set up telemetry collection and viewing using standard OpenTelemetry tooling. Below is an example of a observability setup where Saleor is configured to send telemetry data to an [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/), which then exports the data to a [Jaeger](https://www.jaegertracing.io/) instance for visualization.
+
+### 1. Configure an OpenTelemetry Collector
+
+Set up an OpenTelemetry Collector to receive, process, and export telemetry data.
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:4317
+      http:
+        endpoint: 0.0.0.0:4318
+
+processors:
+  batch:
+
+exporters:
+  jaeger:
+    endpoint: jaeger:4317  # example Jaeger endpoint
+    tls:
+      insecure: true  # enable TLS in production
+
+service:
+  pipelines:
+    traces:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [jaeger]
+```
+
+Note: Only traces are exported in this example as Jaeger doesn't support metrics. You can add additional exporters for metrics if needed.
+
+### 2. Set Up Environment Variables
+
+Configure the necessary environment variables for your Saleor instance to connect to your telemetry system:
+
+```bash
+# OpenTelemetry configuration
+export OTEL_SERVICE_NAME=saleor
```

Comment:
Can you provide a description of the envs? It would give you mroe context than example value like `saleor` or `localhost`.

Reply:
Also, don't you need to mention them here: https://docs.saleor.io/setup/configuration ?

Reply:
That's a good point!

Reply:
I added a link to a page that explains the configuration. Also, I decided not to include it in the `https://docs.saleor.io/setup/configuration` page as these are not Saleor-specific settings. They can be set with a command-line argument as well, and there are many more config options available. I think it's beyond the scope of our docs, and OpenTelemetry docs should be used for that instead.